common/tools.py

```diff
@@ -378,61 +379,64 @@ def get_native_language_and_completeness(language_code):
 # |---------------------------------------|
 
 
-def validate_snapshots_path(path, cfg, profile_id=None):
+def validate_snapshots_path(path: Union[str, pathlib.Path],
+                            host_user_profile: tuple[str, str, str],
+                            mode: str,
+                            copy_links: bool,
+                            error_handler: callable) -> bool:
     """Check if the given path is valid for being a snapshot path.
 
     It is checked if it is a folder, if it is writable, if the filesystem is
     supported and several other things.
 
+    Dev note  (buhtz, 2024-09): That code is a good candidate to get moved
+        into a class or module.
+
     Args:
         path: The path to validate as a snapshot path.
-        cfg: The config instance.
-        profile_id: Related snapshots profile id.
+        host_user_profile: I three item list.
+        mode: The profiles mode.
+        copy_links: The copy links value.
+        error_handler: Handle function receiving error messages.
 
-    Returns:
-        (bool): `False` or `True`
+    Returns: Success (`True`) or failure (`False`).
     """
     if not isinstance(path, pathlib.Path):
         path = pathlib.Path(path)
 
-    if profile_id is None:
-        profile_id = cfg.currentProfile()
-
-    mode = cfg.snapshotsMode(profile_id)
-
     if not path.is_dir():
-        cfg.notifyError(_('Invalid option. {path} is not a folder.')
-                        .format(path=path))
+        error_handler(_('Invalid option. {path} is not a folder.')
+                      .format(path=path))
         return False
 
     # build full path
     # <path>/backintime/<host>/<user>/<profile_id>
-    host, user, profile = cfg.hostUserProfile(profile_id)
-    full_path = path / 'backintime' / host / user / profile
+    host_user_profile = pathlib.Path(os.sep.join(host_user_profile))
+    full_path = path / 'backintime' / host_user_profile
```

Comment:
Could just use `*` unpacking:

``` python
    host_user_profile = pathlib.Path(*host_user_profile)
    full_path = path / 'backintime' / host_user_profile
```

Alternatively, could use `os.sep.join`, but don't need `pathlib.Path`:

``` python
    host_user_profile = os.sep.join(host_user_profile)
    full_path = path / 'backintime' / host_user_profile
```

In fact, pathlib is smart enough that you could just do:

``` python
    full_path = path / 'backintime' / '/'.join(host_user_profile)
```

But the slickest option might be:

``` python
    full_path = pathlib.Path(path, 'backintime', *host_user_profile)
```